src/core/types.ts

```diff
@@ -35,22 +35,34 @@ export type TimestampedMessage = BaseMessage & {
 };
 
 /**
- * Message that references `dataCid`.
+ * Message returned in a query result.
+ * NOTE: the message structure is a modified version of the message received, the most notable differences are:
+ * 1. does not contain `authorization`
+ * 2. may include encoded data
  */
-export type DataReferencingMessage = {
-  descriptor: {
-    dataCid: string;
-  };
-
-  encodedData: string;
+export type QueryResultEntry = {
+  descriptor: Descriptor;
+  encodedData?: string;
 };
 
 export type EqualFilter = string | number | boolean;
 
 export type OneOfFilter = EqualFilter[];
 
-type GT = ({ gt: string } & { gte?: never }) | ({ gt?: never } & { gte: string });
-type LT = ({ lt: string } & { lte?: never }) | ({ lt?: never } & { lte: string });
+
+/**
+ * "greater than" or "greater than or equal to" range condition. `gt` and `gte` are mutually exclusive.
+ */
+export type GT = ({ gt: string } & { gte?: never }) | ({ gt?: never } & { gte: string });
+
+/**
+ * "less than" or "less than or equal to" range condition. `lt`, `lte` are mutually exclusive.
+ */
+export type LT = ({ lt: string } & { lte?: never }) | ({ lt?: never } & { lte: string });
+
+/**
+ * Ranger filter. 1 condition is required.
```

Comment:
🙏 much appreciate all the comments you're adding in this PR

Reply:
Honestly, I didn't understanding exactly the range filter type, but went with 100% trust here.